scripts/clean-db-dump.sql

```diff
@@ -25,10 +25,18 @@ WHERE framework_id IN
      WHERE status='open');
 
 -- Remove data related to open DOS procurements
--- (We can't tell if a procurement is ongoing, so delete all brief responses)
+-- (Delete all brief responses except those with 'awarded' status, where the procurement has ended)
 \echo 'Delete brief responses'
 DELETE
-FROM brief_responses;
+FROM brief_responses
+WHERE awarded_at IS NOT NULL;
+
+-- Sanitise personal data in awarded brief responses
+\echo 'Update brief responses'
+UPDATE brief_responses
+SET data = data::jsonb - 'respondToEmailAddress' || '{"respondToEmailAddress": "cleaned-example-email@gov.uk"}'::jsonb
+WHERE data IS NOT NULL
+AND cast(data AS TEXT) != 'null';
```

Comment:
Presumably this is _necessary_ in order for the update to not fail -- because if it isn't required to be there then I imagine it's just slowing the query down.

Reply:
I copied it from another JSON blob sanitisation below - `signed_agreement_details`.

Reply:
You shouldn't need the `- 'respondToEmailAddress'` the `||` updates existing.

There are 2 different ways of casting used there, the `::JSONB` and the `cast(data AS TEXT)`.

Effectively:
`cast(data AS TEXT) == data::TEXT`
and
`cast(data AS JSONB) == data::jsonb`

We may also need to limit on `data::text != '{}'` IIRC 🤔

Reply:
That makes sense. I'll try and apply the casting consistently. Do you know if we need both `data IS NOT NULL AND cast(data AS TEXT) != 'null'`?

Reply:
Actually I don't _think_ we need either of those.
Our json fields should be `'{}'::json` if empty.